src/app/pages/courses/courses.routes.ts

```diff
@@ -1,10 +1,33 @@
 import { Routes } from '@angular/router';
 
 import { AppRoutePath } from '@app/enums/app-route-path.enum';
+import { AuthGuard } from '@app/guards/auth/auth.guard';
+import { getStaticBreadcrumbsStepsByRoutePath } from '@app/util/util';
 
-import { CourseAddPageComponent, CoursesPageComponent } from '.';
+import { CourseAddPageComponent, CourseEditPageComponent, CoursesPageComponent } from '.';
 
-export const coursesRoutes: Routes = [
-  { path: AppRoutePath.CourseAdd, component: CourseAddPageComponent },
-  { path: AppRoutePath.Courses, component: CoursesPageComponent },
-];
+export const coursesRoutes: Routes = [ {
```

Comment:
here is example of complex routing with public and private views:

`app-routing`:
```typescript

export const routes = [
  {
    path: '',
    children: [
      {
        path: '',
        component: PrivateLayoutComponent,
        children: [
          {
            path: '',
            redirectTo: RoutesPaths.businessManagement,
            pathMatch: 'full',
          },
          {
            path: RoutesPaths.businessManagement,
            canActivate: [ BusinessGuard ],
            loadChildren: () => import('app/business-management/business-management.module').then(m => m.BusinessManagementModule),
          },
          {
            path: RoutesPaths.terms,
            loadChildren: () => import('app/terms-conditions/terms-conditions.module').then(m => m.TermsConditionsModule),
          },
        ],
      },
      {
        path: '',
        canActivate: [ PublicRoutesGuard ],
        children: [
          {
            path: RoutesPaths.error,
            loadChildren: () => import('app/_errors/errors.module').then(m => m.ErrorsModule),
          },
          {
            path: RoutesPaths.asm, // same as login
            resolve: {
              queryParams: AsmPageQueryParamsResolver,
            },
            canActivate: [ AsmGuard ],
            loadChildren: () => import('app/asm/asm.module').then(m => m.AsmModule),
          },
        ],
      },
      {
        path: '**',
        redirectTo: '',
      },
    ],
  },

  /**
   * Please keep type definition with `as` syntax.
   *
   * Work around for resolving an issue during the compodoc generation
   * https://github.com/compodoc/compodoc/issues/525
   */
] as Routes;

``

child routing
```typescript

const routes: Routes = [ {
  path: '',
  data: {
    breadcrumbs: 'navigation.business-management',
  },
  children: [
    {
      path: '',
      redirectTo: RoutesPaths.myPerformance,
      pathMatch: 'full',
    },
    {
      path: RoutesPaths.myPerformance,
      loadChildren: () => import('app/my-performance/my-performance.module').then(m => m.MyPerformanceModule),
      data: {
        allowedRoles: [
          ...getAllowedRolesByPath(RoutesPaths.myPerformance, APP_NAVIGATION_CONFIG),
        ],
      },
    },
  ],
} ];
```

Reply:
Thanks